packages/http/src/mocker/generator/__tests__/HttpParamGenerator.spec.ts

```diff
@@ -0,0 +1,53 @@
+import { assertNone, assertSome } from '@stoplight/prism-core/src/utils/__tests__/utils';
+import { HttpParamStyles } from '@stoplight/types/dist';
+import { generate } from '../HttpParamGenerator';
+
+describe('HttpParamGenerator', () => {
+  describe('generate()', () => {
+    describe('example is present', () => {
+      it('uses static example', () => {
+        assertSome(generate({ name: 'a', style: HttpParamStyles.Form, examples: [{ key: 'foo', value: 'test' }] }), v =>
+          expect(v).toEqual('test'),
+        );
+      });
+    });
+
+    describe('schema and example is present', () => {
+      it('prefers static example', () => {
+        assertSome(
+          generate({
+            name: 'a',
+            style: HttpParamStyles.Form,
+            schema: { type: 'string' },
+            examples: [{ key: 'foo', value: 'test' }],
+          }),
+          v => expect(v).toEqual('test'),
+        );
+      });
+    });
+
+    describe('schema is present', () => {
+      it('generates example from schema', () => {
+        assertSome(
+          generate({
+            name: 'a',
+            style: HttpParamStyles.Form,
+            schema: { type: 'string' },
+          }),
+          v => expect(v).toEqual(expect.any(String)),
```

Comment:
Can we use the `format` schema parameter so we can narrow down the generated string so we can check out the schema is being used for real?

Reply:
Yes, I'd really say let's use the format keyword and try to generate an email or something.